test/catch.cc

```diff
@@ -86,12 +86,39 @@ TEST(CatchTest, All) {
                  ADD_FAILURE() << "Encountered unexpected matched raised";
                  return 10;
                })
-               .all([](std::exception_ptr&& error) {
-                 try {
-                   std::rethrow_exception(error);
-                 } catch (const std::runtime_error& error) {
-                   EXPECT_STREQ(error.what(), "10");
-                 }
+               .all([](std::variant<std::runtime_error>&& error) {
+                 EXPECT_STREQ(std::get<0>(error).what(), "10");
```

Comment:
No more please unless we use explicit indexing like in `do-all.h`!!!
```suggestion
                 EXPECT_STREQ(std::get<std::runtime_error>(error).what(), "10");
```